### PR TITLE
GH-703 Update to Theia 1.27.0

### PR DESCRIPTION
**Pull request by tortmayr**
- Update Theia dependencies to 1.27.0
- Adapt code base to API breaks of 1.27.0
- Implement a temporary workaround for tunneling json-rpc connection until https://github.com/eclipse-theia/theia/issues/11405 is resolved
- Update readmes

Fixes https://github.com/eclipse-glsp/glsp/issues/703

Contributed on behalf of STMicroelectronics

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
 # Eclipse GLSP Theia Integration Changelog
 
+## [Upcoming - TBA]()
+
+### Breaking Changes
+
+-   [theia] Updated Theia dependencies to `1.27.0`. Due API breaks Theia version `<1.27.0` are no longer supported. [#105](https://github.com/eclipse-glsp/glsp-theia-integration/pull/105) - Contributed on behalf of STMicroelectronics <br>
```
Comment (ndoschek) [inline CHANGELOG.md:7]:
```suggestion
-   [theia] Updated Theia dependencies to `1.27.0`. Due to API breaks, Theia versions `<1.27.0` are no longer supported. [#105](https://github.com/eclipse-glsp/glsp-theia-integration/pull/105) - Contributed on behalf of STMicroelectronics <br>
```

File: packages/theia-integration/src/browser/channel-connection.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
+/********************************************************************************
+ * Copyright (c) 2022 EclipseSource and others.
```
Comment (ndoschek) [inline packages/theia-integration/src/browser/channel-connection.ts:2]: Perhaps?
 
```suggestion
 * Copyright (c) 2022 STMicroelectronics and others.
```

File: packages/theia-integration/src/node/connection-forwarder.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
+/********************************************************************************
+ * Copyright (c) 2022 EclipseSource and others.
```
Comment (ndoschek) [inline packages/theia-integration/src/node/connection-forwarder.ts:2]: Same here

**Review by ndoschek — CHANGES_REQUESTED** [review]
Thanks Tobias, the fix looks great to me :tada:! I tested with the current `workflow` example and everything seems to work again as expected!
I added a few minor nitpicks (please see inline).

Also, in the GLSP server log I noticed the following `protocolVersion=0.9.0`:
```
2022-07-11 15:43:31,599422000 [pool-2-thread-1] DEBUG DefaultGLSPServer - Initializing server with the following params:
InitializeParameters [applicationId=fe1c055b-1242-492c-af85-cb845bf55c98, protocolVersion=0.9.0, args={ timestamp=Mon Jul 11 2022 15:43:30 GMT+0200 (Central European Summer Time),message=Custom Options Available } ]
```
This should be updated as well I assume?

**Comment by tortmayr** [comment]
> Thanks Tobias, the fix looks great to me tada! I tested with the current `workflow` example and everything seems to work again as expected! I added a few minor nitpicks (please see inline).
> 
> Also, in the GLSP server log I noticed the following `protocolVersion=0.9.0`:
> 
> ```
> 2022-07-11 15:43:31,599422000 [pool-2-thread-1] DEBUG DefaultGLSPServer - Initializing server with the following params:
> InitializeParameters [applicationId=fe1c055b-1242-492c-af85-cb845bf55c98, protocolVersion=0.9.0, args={ timestamp=Mon Jul 11 2022 15:43:30 GMT+0200 (Central European Summer Time),message=Custom Options Available } ]
> ```
> 
> This should be updated as well I assume?

Probably, yes. However, this also requires a server change. I suppose it would be better if we handle is in a follow-up PR.
I will create an issue for that

**Comment by tortmayr** [comment]
Thanks Nina for the fast review. I have addressed your comments and have created an issue for the protocol version mismatch (https://github.com/eclipse-glsp/glsp/issues/706)

**Review by ndoschek — APPROVED** [review]
Thanks again, LGTM :+1:

**Pull request merged**